tex/generic/pgf/libraries/shapes/pgflibraryshapes.multipart.code.tex

```diff
@@ -300,6 +300,119 @@
 }%
 
 
+%
+% A diamond that is split in the middle into an upper and a lower part.
+%
+% This node consists of two parts: the upper (text) part is shown in the
+% upper half of the diamond. The second part is the (optional) lower part.
+%
+% Parts: text, lower
+
+\pgfdeclareshape{diamond split}
+{%
+  %
+  % Node parts
+  %
+  \nodeparts{text,lower}
+  %
+  % Anchors
+  %
+  \savedanchor\outernortheast{%
+    %
+    % Calculate width and height of the text rectangle
+    %
+    \pgf@xa=\wd\pgfnodeparttextbox%
+    \pgfmathsetlength\pgf@xc{\pgfkeysvalueof{/pgf/inner xsep}}%
+    \advance\pgf@xa by 0\pgf@xc%
+    \pgf@ya=\ht\pgfnodeparttextbox%
+    \advance\pgf@ya by \dp\pgfnodeparttextbox%
+    \pgfmathsetlength\pgf@yc{\pgfkeysvalueof{/pgf/inner ysep}}%
+    \advance\pgf@ya by 0\pgf@yc%
+    %
+    % (xa,ya) is now the size of the *text* part.
+    %
+    % Calculate width and height of the lower rectangle
+    %
+    \pgf@xb=\wd\pgfnodepartlowerbox%
+    \advance\pgf@xb by \pgf@xc%
+    \pgf@yb=\ht\pgfnodepartlowerbox%
+    \advance\pgf@yb by \dp\pgfnodepartlowerbox%
+    \advance\pgf@yb by \pgf@yc%
+    %
+    % (xb,yb) is now the size of the *lower* part.
+    % We keep the maximum value of xa vs. xb and ya vs. yb
+    %
+    \pgfmathsetlength{\pgf@xa}{max(\pgf@xa,\pgf@xb)}
+    \pgfmathsetlength{\pgf@ya}{max(\pgf@ya,\pgf@yb)}
```

Comment:
No reason to run the full parsing machinery for something that you know is only a single operation on two numbers.
```suggestion
    \pgfmathmax@{{\pgfmath@tonumber{\pgf@xa}}{\pgfmath@tonumber{\pgf@xb}}}%
    \pgf@xa=\pgfmathresult pt
    \pgfmathmax@{{\pgfmath@tonumber{\pgf@ya}}{\pgfmath@tonumber{\pgf@yb}}}%
    \pgf@ya=\pgfmathresult pt
```